Initial A71 support

pascua28 opened the pull request:
TODO:

- [x] Replace as much as possible the a52q prebuilts

Bugs:

- ~~Display delay due to new display stack (caused by sync timeout from BufferQueueProducer)~~

pascua28 committed:
prebuilts/samsung: a73xqxx: add vintf manifest files

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: a73xqxx: add vintf manifest files

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: a73xqxx add vibrator blobs

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: a73xqxx add vibrator blobs

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: a73xqxx: add more camera libs

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: a73xqxx: add more camera libs

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: display: add patched qdcm calibration

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: display: add patched qdcm calibration

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: hex-patch sdmutils WaitSync() timeout

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: hex-patch sdmutils WaitSync() timeout

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: add r0qksx prebuilts

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: add r0qksx prebuilts

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

github/workflows: change r0qksx update CSC

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: add m36xxx prebuilts

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: add m36xxx prebuilts

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

Fede2782 commented on target/a71/sff.sh:
Let the build system automatically do floating_feature, then diff stock one with the generated one to create this file to keep it more clean.

pascua28 replied:
Done (https://github.com/salvogiangri/UN1CA/pull/528/commits/c56fa8ccc9a6a66ae093750911ce7160ffa46c94). Thanks

Fede2782 commented on target/a71/patches/mainline-fix/module.prop:
Is this still required? if you are using a custom kernel it should already support proper BPF and spoofing. This was intended for devices like S20 FE whose stock kernel was missing the BPF JIT config (enforced later by a mainline update). Now it is required stock 5.4 or later kernel, which should work as is, or a custom one which should have BPF backports and spoof.

pascua28 replied:
Hi, yes. For some reason, even with backported 5.4 bpf, when installing new Google play system update, the phone still bootloops and rolls back the update.

pascua28 committed:
prebuilts/samsung: a73xqxx: add display stack

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
prebuilts/samsung: a73xqxx: add display stack

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

salvogiangri committed:
a71: init

Co-authored-by: Samuel Pascua <pascua.samuel.14@gmail.com>
Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

salvogiangri committed:
a71: init

Co-authored-by: Samuel Pascua <pascua.samuel.14@gmail.com>
Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: fix `GetCustomDimensions()` error

Fix `GetCustomDimensions()` error logspam from graphics mapper caused by
early exit after calling `getMetaData()` especially when using newer
adreno drivers.

No functional change.

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: fix `GetCustomDimensions()` error

Fix `GetCustomDimensions()` error logspam from graphics mapper caused by
early exit after calling `getMetaData()` especially when using newer
adreno drivers.

No functional change.

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: fetch kernel files from UN1CA/kernel_samsung_sm7150

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

aurafih commented:
Hey pascua28! I’m a UN1CA user on a Galaxy A71 (A715F). I’m familiar with basic installing and debugging, but I’m not an expert on the development side. I was just wondering: is UN1CA 3.0 (One UI 8) planned for the A71?

pascua28 commented:
> Hey pascua28! I’m a UN1CA user on a Galaxy A71 (A715F). I’m familiar with basic installing and debugging, but I’m not an expert on the development side. I was just wondering: is UN1CA 3.0 (One UI 8) planned for the A71?

There's already a public test build for A71 based on OneUI 8

pascua28 committed:
a71: Disable HeatMapThread logspam

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Disable HeatMapThread logspam

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Update vintf FCM to level 5

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Update vintf FCM to level 5

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Enable extra brightness feature

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Enable extra brightness feature

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

majaahh commented on target/a71/patches/camera/customize.sh:
```diff
@@ -0,0 +1,93 @@
+ADD_TO_WORK_DIR "m36xxx" "system" "system/lib64/libPortraitSolution.camera.samsung.so" 0 0 644 "u:object_r:system_lib_file:s0"
+LOG_STEP_OUT
+
+if ! grep -q "Camera End" "$WORK_DIR/vendor/ueventd.rc"; then
```
No need to keep that check. This was required back with the old build system.

majaahh commented on target/a71/patches/camera/customize.sh:
```diff
@@ -0,0 +1,93 @@
+
+if ! grep -q "Camera End" "$WORK_DIR/vendor/ueventd.rc"; then
+    echo "" >> "$WORK_DIR/vendor/ueventd.rc"
+    cat "$SRC_DIR/target/a71/patches/camera/ueventd" >> "$WORK_DIR/vendor/ueventd.rc"
```
```suggestion
    {
        echo ""
        cat "$MODPATH/ueventd"
    } >> "$WORK_DIR/vendor/ueventd.rc"
```

pascua28 replied:
637ce5004e9e7301c7cba44fe8000bcc784d236c

majaahh commented on target/a71/patches/camera/customize.sh:
```diff
@@ -0,0 +1,93 @@
+"
+for blob in $BLOBS_LIST
+do
+    DELETE_FROM_WORK_DIR "system" "$blob"
```
```suggestion
for i in $BLOBS_LIST; do
    DELETE_FROM_WORK_DIR "system" "$i"
```

majaahh commented on target/a71/patches/camera/customize.sh:
```diff
@@ -0,0 +1,93 @@
+"
+for blob in $BLOBS_LIST
+do
+    ADD_TO_WORK_DIR "a73xqxx" "system" "$blob" 0 0 644 "u:object_r:system_lib_file:s0"
```
```suggestion
for i in $BLOBS_LIST; do
    ADD_TO_WORK_DIR "a73xqxx" "system" "$i 0 0 644 "u:object_r:system_lib_file:s0"
```

pascua28 replied:
637ce5004e9e7301c7cba44fe8000bcc784d236c

majaahh commented on target/a71/patches/kernel/customize.sh:
```diff
@@ -0,0 +1,8 @@
+
+LOG "- Downloading Paradigm kernel"
+[ -f "$WORK_DIR/kernel/boot.img" ] && rm -f "$WORK_DIR/kernel/boot.img"
+[ -f "$WORK_DIR/kernel/dtbo.img" ] && rm -f "$WORK_DIR/kernel/dtbo.img"
```
https://github.com/salvogiangri/UN1CA/commit/75af9a2db13ae87d5d0ad5c4146bc0b954a7abc7

majaahh commented on target/a71/patches/kernel/customize.sh:
```diff
@@ -0,0 +1,8 @@
+KERNEL_REPO="https://github.com/UN1CA/kernel_samsung_sm7150/releases/latest/download"
```
Unset at the end of the file

majaahh commented on target/a71/patches/stock_blobs/customize.sh:
```diff
@@ -0,0 +1,108 @@
+
+LOG_STEP_IN "- Fixing MIDAS model detection"
+sed -i "s/a73xq/a71/g" "$WORK_DIR/vendor/etc/midas/midas_config.json"
+sed -i "s/ro.product.device/ro.product.vendor.device/g" "$WORK_DIR/vendor/etc/midas/midas_config.json"
```
Use EVAL

majaahh commented on target/a71/patches/stock_blobs/customize.sh:
```diff
@@ -0,0 +1,108 @@
+LOG_STEP_OUT
+
+LOG_STEP_IN "- Fixing RIL"
+sed -i "s/1.4::IRadio/1.5::IRadio/g" "$WORK_DIR/vendor/etc/vintf/manifest.xml"
```
Use EVAL

majaahh commented on target/a71/patches/stock_blobs/file_context-system:
```diff
@@ -0,0 +1,16 @@
+/ u:object_r:rootfs:s0
```
Unnecessary, use ```/system u:object_r:system_file:s0``` instead

pascua28 replied:
637ce5004e9e7301c7cba44fe8000bcc784d236c

majaahh commented on target/a71/patches/vibrator/customize.sh:
```diff
@@ -0,0 +1,13 @@
+DELETE_FROM_WORK_DIR "vendor" "lib64/vendor.samsung.hardware.vibrator@2.0.so"
+DELETE_FROM_WORK_DIR "vendor" "lib64/vendor.samsung.hardware.vibrator@2.1.so"
+DELETE_FROM_WORK_DIR "vendor" "lib64/vendor.samsung.hardware.vibrator@2.2.so"
+sed -i '/<hal format="hidl">.*/{:a;N;/<\/hal>/!ba;/android.hardware.vibrator/d}' $WORK_DIR/vendor/etc/vintf/manifest.xml
```
Use EVAL

majaahh commented on target/a71/sff.sh:
```diff
@@ -0,0 +1,31 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
```
https://github.com/salvogiangri/UN1CA/commit/85050ae5542a7d78802548a1fd0ef4d24517fcf7

pascua28 replied:
7321e6b75468648dfbefad98555e4ee3d64819fb

majaahh commented on target/a71/debloat.sh:
```diff
@@ -0,0 +1,76 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
```
https://github.com/salvogiangri/UN1CA/commit/85050ae5542a7d78802548a1fd0ef4d24517fcf7

majaahh commented on target/a71/config.sh:
```diff
@@ -0,0 +1,89 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
```
https://github.com/salvogiangri/UN1CA/commit/85050ae5542a7d78802548a1fd0ef4d24517fcf7

Fede2782 commented on target/a71/sff.sh:
```diff
@@ -24,8 +24,5 @@ SEC_FLOATING_FEATURE_LCD_CONFIG_HFR_SUPPORTED_REFRESH_RATE_NS=
 SEC_FLOATING_FEATURE_CAMERA_CONFIG_VENDOR_LIB_INFO=image_codec.samsung.v1,beauty.samsung.v4,face_landmark.arcsoft.v2_1,facial_attribute.samsung.v1,swuwdc.arcsoft.v1,event_detection.samsung.v2,food.samsung.v1,selfie_correction.samsung.v1,dual_bokeh.samsung.v1,single_bokeh.samsung.v2,image_enhance.arcsoft.v1,mfhdr.arcsoft.v1,llhdr.arcsoft.v1,scene_detection.samsung.v1,smart_scan.samsung.v2,human_tr ...
 
-# Samsung Gallery app
-SEC_FLOATING_FEATURE_GALLERY_CONFIG_IMAGE_TAGGER_VERSION=V701
```
You need to add classifier blobs in both system/vendor for full image tagger. Also To get v701 image tagger you also need to manually add the library from another device and making sure your blobs are update enough.

pascua28 replied:
https://github.com/salvogiangri/UN1CA/pull/528/changes#diff-910cb9cc471a9575e5916e07f890302e42d1d9bbf1307e1aba7f36889fcf7f02R17

Tagger lib is from a73: https://github.com/salvogiangri/UN1CA/pull/528/changes#diff-a3d44003b90c09c93d58a382ecbd83cd58e5844a780ed38c4223466f695ea6e9R63 which is v701.

pascua28 committed:
unica: patches: nfc: Add NXP_PN553 support

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

ata-kaner committed:
prebuilts: r9qxxx: Add Sec RIL blobs

Change-Id: I750ca648365224dc6d46f3bb002844c0b76d0620

pascua28 committed:
a71: switch to SPDX license headers

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: switch to SPDX license headers

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: refactor code

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: refactor code

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: update partition configs

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: update partition configs

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: add stock recovery.fstab

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: add stock recovery.fstab

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Fix ISehBiometricsFace version

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Fix ISehBiometricsFace version

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

majaahh commented on target/a71/debloat.sh:
```diff
@@ -0,0 +1,62 @@
+system/etc/permissions/privapp-permissions-com.samsung.android.aremojieditor.xml
+system/etc/permissions/privapp-permissions-com.sec.android.mimage.avatarstickers.xml
+system/priv-app/AREmojiEditor
+system/priv-app/AvatarEmojiSticker
```
Debloated by UN1CA's debloat script

pascua28 replied:
4bfd952f4fbd080dbb2cd59a918e0c679d649ba5

majaahh commented on target/a71/debloat.sh:
```diff
@@ -0,0 +1,62 @@
+system/priv-app/AREmojiEditor
+system/priv-app/AvatarEmojiSticker
+system/priv-app/EarphoneTypeC
+system/priv-app/IntelligentDynamicFpsService
```
Debloated by UN1CA's debloat script if TARGET_LCD_CONFIG_HFR_MODE is higher than 1

majaahh commented on target/a71/debloat.sh:
```diff
@@ -0,0 +1,62 @@
+system/priv-app/AvatarEmojiSticker
+system/priv-app/EarphoneTypeC
+system/priv-app/IntelligentDynamicFpsService
+system/priv-app/SohService
```
Auto removed by unica/patches/stock_blobs/customize.sh if it doesn't exist in stock firmware

majaahh commented on target/a71/patches/stock_blobs/customize.sh:
```diff
@@ -0,0 +1,114 @@
+ADD_TO_WORK_DIR "$TARGET_FIRMWARE" "system" "system/lib64/libsamsungSoundbooster_plus_legacy.so" 0 0 644 "u:object_r:system_lib_file:s0"
+LOG_STEP_OUT
+
+LOG_STEP_IN "- Fixing RIL"
```
```suggestion
LOG "- Replacing every occurrence of \"1.4::IRadio\" with \"1.5::IRadio\" in /vendor/etc/vintf/manifest.xml
```

pascua28 replied:
b8b138896db409ec290a25611d9754e9043d34ac

majaahh commented on target/a71/patches/vibrator/customize.sh:
```diff
@@ -0,0 +1,13 @@
+DELETE_FROM_WORK_DIR "vendor" "lib64/vendor.samsung.hardware.vibrator@2.0.so"
+DELETE_FROM_WORK_DIR "vendor" "lib64/vendor.samsung.hardware.vibrator@2.1.so"
+DELETE_FROM_WORK_DIR "vendor" "lib64/vendor.samsung.hardware.vibrator@2.2.so"
+EVAL "sed -i '/<hal format=\"hidl\">.*/{:a;N;/<\/hal>/!ba;/android.hardware.vibrator/d}' \"$WORK_DIR/vendor/etc/vintf/manifest.xml\""
```
I'd add a ```LOG``` for that that matches log commands style used in ```SMALI_PATCH``` if possible

pascua28 replied:
b8b138896db409ec290a25611d9754e9043d34ac

pascua28 committed:
a71: update debloat list

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: update debloat list

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Improve logging

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Improve logging

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Enable AVB signing

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: add FBEv2 support

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

pascua28 committed:
a71: Update RIL from r9q

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

Fede2782 commented on target/a71/debloat.sh:
```diff
@@ -0,0 +1,51 @@
+
+# Gallery AI Semantic Search
+SYSTEM_DEBLOAT+="
+system/etc/default-permissions/default-permissions-com.samsung.mediasearch.xml
```
Do we really need to debloat this? If we need space there are probably other ways to free up space without sacrificing useful features

Fede2782 commented on target/a71/debloat.sh:
```diff
@@ -0,0 +1,51 @@
+
+# Apps debloat
+PRODUCT_DEBLOAT+="
+priv-app/HotwordEnrollmentOKGoogleEx4HEXAGON
```
A71 doesn't have a stock hotword enrollment?

Fede2782 commented:
Hi everyone! Since I believe this is getting close to merge and I have seen @salvogiangri testing it as well, can the main maintainer also ensure all the hardware and software works fine?

If possible would be great to ensure full compliance with the UN1CA CDD: https://github.com/Fede2782/UN1CA/wiki/UN1CA-Compatibility-Definition-Document

In case of any problem while applying the criteria on this device, I would really appreciate any feedback regarding to it.

Thanks for your time!

salvogiangri reviewed (approved):
LGTM
